### PR TITLE
Avoid telemetry cluster metadata override

### PR DESCRIPTION
**Pull request by bianpengyuan**
Fixes https://github.com/istio/istio/issues/28140

telemetry metadata is overridden by configuration metadata.

### CODE REVIEW

File: pilot/pkg/networking/util/util.go
Comment (kyessenov) [inline pilot/pkg/networking/util/util.go:393]: You could do pointer-to-pointer here and not return.

**Review by howardjohn — APPROVED** [review]

File: pilot/pkg/networking/util/util.go
Comment (bianpengyuan) [reply pilot/pkg/networking/util/util.go:393]: I tried that out, but does not seem to be intuitive comparing to the current code.

Comment (kyessenov) [reply pilot/pkg/networking/util/util.go:393]: Yeah, I figured it. Requires some C++ mindset :)

**Comment by bianpengyuan** [comment]
@istio/wg-policies-and-telemetry-maintainers ptal, needs approval on this. Thanks!

**Review by kyessenov — APPROVED** [review]

**Comment by bianpengyuan** [comment]
Test failure was due to change of inbound cluster name. Merge upstream and retest..

**Pull request merged**
